addons/calendar_bis/models/calendar_event.py

```diff
@@ -0,0 +1,137 @@
+from odoo import models, fields, api, _, Command
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import UserError
+
+from ..util import RRULE
+
+recurring_fields = ['is_recurring', 'mo', 'tu', 'we', 'th', 'fr', 'sa', 'su', 'freq', 'until', 'count', 'interval',
+                    'monthday', 'monthweekday_n', 'monthweekday_day']
+
+
+class CalendarEventPrivate(models.Model):
+    _name = "calendar.event"
+    _description = "Calendar Event Private"
+    _order = "id desc"
+
+    def _valid_field_parameter(self, field, name):  # USED FOR GENERATION OF THE PUBLIC FILE, REMOVE WHEN FINALIZED
+        return name in ['public', 'public_default'] or super()._valid_field_parameter(field, name)
+
+    timeslot_ids = fields.One2many('calendar.timeslot', 'event_id')
+
+    is_public = fields.Boolean(default=False, public=True)
+    partner_id = fields.Many2one('res.partner', string="Owner", public=True, default=lambda self: self.env.user.partner_id.id)
+
+    name = fields.Char(public_default="Busy")
+    note = fields.Char()
+
+    # Recurrence
+    is_recurring = fields.Boolean(default=False, public=True)
```

Comment:
for now, the recurrence is handled by several fields:
recurrency: bool to know if the event is in a recurrence
recurrence_id: the linked recurrence
follow_recurrence: to take into account events that does not follow the rule. You use exdated for that purpose I think. But you don't have an object to store the recurrence. I think your model is closer to what microsoft or google api are sending.